src/main/java/org/fenixedu/academic/domain/EnrolmentPeriod.java

```diff
@@ -83,6 +83,10 @@ protected void init(final DegreeCurricularPlan degreeCurricularPlan, final Execu
         setEndDateDateTime(endDate);
     }
 
+    public abstract boolean isForCurricularCourses();
```

Comment:
I suggest having them both return `false`, and only override to `true` where needed.